examples/tic-tac-toe/cli/src/client.rs

```diff
@@ -0,0 +1,749 @@
+        let opponent = SuiAddress::from(&opponent_key);
+
+        // A 1-of-2 multisig acts as the admin of the game. The Game object will be transferred to
+        // this address once it is created.
+        let admin_key = combine_keys(vec![player_key, opponent_key])?;
+        let admin = SuiAddress::from(&admin_key);
+        let admin_bytes = bcs::to_bytes(&admin_key).context("Encoding admin key")?;
+
+        let mut builder = ProgrammableTransactionBuilder::new();
+        let x = builder.pure(player)?;
+        let o = builder.pure(opponent)?;
+        let a = builder.pure(admin_bytes)?;
+
+        let game = builder.programmable_move_call(
+            self.package,
+            Identifier::new("owned").unwrap(),
+            Identifier::new("new").unwrap(),
+            vec![],
+            vec![x, o, a],
+        );
+
+        builder.transfer_arg(admin, game);
+
+        let tx = self.build_tx_data(player, builder.finish()).await?;
+        self.execute_for_game(tx).await
+    }
+
+    /// Delete a shared game, given itself contents and its ownership information (which should be a
+    /// `Owner::Shared`).
+    pub async fn delete_shared_game(&mut self, game: &game::Shared, owner: Owner) -> Result<()> {
+        let player = self.wallet.active_address()?;
+
+        let Owner::Shared {
+            initial_shared_version,
+        } = owner
+        else {
+            bail!("Game is not shared");
+        };
+
+        let mut builder = ProgrammableTransactionBuilder::new();
+
+        let g = builder.obj(ObjectArg::SharedObject {
+            id: game.board.id,
+            initial_shared_version,
+            mutable: true,
+        })?;
+
+        builder.programmable_move_call(
+            self.package,
+            Identifier::new("shared").unwrap(),
+            Identifier::new("burn").unwrap(),
+            vec![],
+            vec![g],
+        );
+
+        let data = self.build_tx_data(player, builder.finish()).await?;
+        let tx = self.wallet.sign_transaction(&data);
+        self.execute_transaction(tx).await?;
+        Ok(())
+    }
+
+    /// Delete an owned (multi-sig) game. The transaction is signed by the player on behalf of the
+    /// admin (multi-sig) address, and also directly by the player who is acting as the sponsor.
+    pub async fn delete_owned_game(
+        &mut self,
+        game: &game::Owned,
+        game_ref: ObjectRef,
+    ) -> Result<()> {
+        let player = self.wallet.active_address()?;
+
+        let mut builder = ProgrammableTransactionBuilder::new();
+
+        let g = builder.obj(ObjectArg::ImmOrOwnedObject(game_ref))?;
+
+        builder.programmable_move_call(
+            self.package,
+            Identifier::new("owned").unwrap(),
+            Identifier::new("burn").unwrap(),
+            vec![],
+            vec![g],
+        );
+
+        let admin_key: MultiSigPublicKey =
+            bcs::from_bytes(&game.admin).context("Failed to deserialize admin's public key.")?;
```

Comment:
nit: in some places the string message has a `.` in other doesn't. Keep it consistent.